src/osd/OSD.cc

```diff
@@ -6539,10 +6643,13 @@ void OSD::handle_osd_map(MOSDMap *m)
 
   if (m->newest_map && m->newest_map > last) {
     dout(10) << " msg say newest map is " << m->newest_map << ", requesting more" << dendl;
-    osdmap_subscribe(osdmap->get_epoch()+1, true);
+    osdmap_subscribe(osdmap->get_epoch()+1, false);
   }
-  else if (is_booting()) {
-    start_boot();  // retry
+  else if (is_preboot()) {
+    if (m->get_source().is_mon())
+      _preboot(m->oldest_map, m->newest_map);
```

Comment:
oldest_map is the one I'm concerned about, not newest. That's filled in to the newer of our oldest map, or the monitor's oldest map, but we treat it as the monitor's oldest in _preboot. (Granted it looks to be mainly debug output in there, but it will be confusing. I had it in my head we were doing some osdmap trims based on these values but now I'm not seeing it, so not sure if there's a more important use or not.)